Makefile.core.mk

```diff
@@ -133,15 +112,22 @@ GO_FILES_CMD := find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)'
 # Environment for tests, the directory containing istio and deps binaries.
 # Typically same as GOPATH/bin, so tests work seemlessly with IDEs.
 
-export ISTIO_BIN=$(GO_TOP)/bin
+export ISTIO_BIN=$(GOBIN)
 # Using same package structure as pkg/
-export OUT_DIR=$(GO_TOP)/out
-export ISTIO_OUT:=$(OUT_DIR)/$(GOOS)_$(GOARCH)/$(BUILDTYPE_DIR)
-export ISTIO_OUT_LINUX:=$(OUT_DIR)/linux_amd64/$(BUILDTYPE_DIR)
-export HELM=$(ISTIO_OUT)/helm
+
+export ISTIO_OUT:=$(TARGET_OUT)
+export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
+export HELM=helm
 export ARTIFACTS ?= $(ISTIO_OUT)
+export JUNIT_OUT ?= $(ARTIFACTS)/junit.xml
 export REPO_ROOT := $(shell git rev-parse --show-toplevel)
 
+# Make directories needed by the build system
+$(shell mkdir -p $(ISTIO_OUT))
+$(shell mkdir -p $(ISTIO_OUT_LINUX))
+$(shell mkdir -p $(ISTIO_OUT_LINUX)/logs)
+$(shell mkdir -p $(dir $(JUNIT_OUT)))
```

Comment:
should this be a target so its not run every time (?)

nice to have: `make -f Makefile.core.mk` doesn't work right now since we have the fields set in Makefile. I think we have some scripts that run this (and I do as well sometimes -- but I can always change my alias)

Reply:
The downside of this logic inn a target is it must be added all over the place in the Makefiles which is fairly annoying to track down and keep correct. The cost of of this logic is very low - even when multiplied by 10s of thousands of runs.

Re `make -f Makefile.core.mk` I could duplicate the environment variables used in the `Makefile` into `Makefile.core.mk` but this doesn't seem resilient to changes in the top level Makefile. A different option would be to put the environment variables in a different Makefile such as `Makefile.env.vars` although this starts to spread out the logic of the common files beyond the override file. I'll give it some thought.

In the short term, I'd like to merge this once things are functional and we can iterate in common-files repo from there on gaps.